dashboard/app/views/home/index.html.haml

```diff
@@ -25,6 +25,10 @@
   - if current_user
     #user_hero
       = render template: 'api/user_hero'
+    - [current_user.working_on_user_scripts, current_user.completed_user_scripts].compact.reduce([], :|).each_slice(2) do |course_pair|
```

Comment:
Nah, I don't want to use flatten because I don't want to break down lists within it, just keep the same structure and combine. I still think reduce is the best way, but I may be wrong.